sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ColumnVector.java

```diff
@@ -847,6 +862,11 @@ public final int appendStruct(boolean isNull) {
   protected int capacity;
 
   /**
+   * Upper limit for the maximum capacity for this column.
+   */
+  protected int MAX_CAPACITY = Integer.MAX_VALUE;
```

Comment:
Nit: This is for testing purposes only right. We have an annotation for that, if I am not mistaking.